src/Sanitizer.php

```diff
@@ -213,4 +290,35 @@ private function remove_js_tags( string $content ): string {
 
 		return (string) preg_replace( '/<script[^>]*>.*?<\/script>/sm', '', $content );
 	}
+
+	/**
+	 * Remove scripts from HTML.
+	 *
+	 * Use DomDocument to remove scripts from HTML.
+	 *
+	 * @since 1.5.0
+	 *
+	 * @param string $content File content.
+	 *
+	 * @return string
+	 */
+	private function remove_scripts_from_html( string $content ): string {
+
+		$dom = new DOMDocument();
```

Comment:
to still have SVG file not be destroyed by sanitizer, I had to whitelist all elements and attributes except `script` above. 

I should to the same for HTML files but list of elements and attributes is simply too huge. I tried to base on `wp_kses` allowed list for post content + elements like `html`, `head`, `body`, `meta` but still was making HTML files changed to non acceptable level. 

So here I did other approach. Blacklisted script tag but delegated the logic to battlefield tested DomDocument library. It traverses dom document and if some element is a script, it removes it.  


(Tbh we could do the same for SVG but use `loadXML`, not `loadHTML` and we will not have to do those tricks to preserve `<?xml` and overcoming wp_kses bug with CDATA

Reply:
There are many ways for XSS payloads (e.g. javascript) to get injected. A personal favorite is <img src=x onerror=alert(1)> which will likely bypass this "remove_scripts_from_html()" function.

You may also review the following URL for many more example XSS payloads https://github.com/payloadbox/xss-payload-list

This is why it is important to look into allow-lists, wp_kses can do this.

You should be able to use an AI agent to help save you time building the list of allowable HTML tags and attributes. Both antropic and google gemini should give you accurate answers.